### PR TITLE
(DOP-2600) Directive for cta-banner

### PR DESCRIPTION
**Pull request by gjchong25**

### CODE REVIEW

File: snooty/rstspec.toml [hunk 1]
```diff
@@ -86,6 +86,15 @@ example = """.. warning:: ${1:Title}
    ${2:Warning block content}
 """
 
+[directive.cta-banner]
```
Comment (i80and) [inline snooty/rstspec.toml:89]: Could we put this into the `mongodb` domain? i.e. `[directive."mongodb:cta-banner"]`

File: snooty/rstspec.toml [hunk 2]
```diff
@@ -4,7 +4,7 @@ version = 0
 [enum]
 alignment = ["left", "center", "right"]
 backlinks = ["entry", "top", "none"]
-banner_variant = ["info", "warning", "danger"]
+banner_variant = ["info", "warning", "danger", "cta-banner"]
```
Comment (i80and) [inline snooty/rstspec.toml:7]: To clarify, this permits:
```
.. banner::
   :variant: cta-banner
```

this is intentional? it's fine if it is, just want to check!

**Review by i80and — CHANGES_REQUESTED** [review]
dynamite stuff

File: snooty/rstspec.toml (hunk 2 above)
Comment (gjchong25) [reply snooty/rstspec.toml:7]: Nope, removed it!

File: snooty/rstspec.toml (hunk 1 above)
Comment (gjchong25) [reply snooty/rstspec.toml:89]: Yep!

File: snooty/parser.py [hunk 3]
```diff
@@ -608,6 +609,12 @@ def handle_directive(
 
             self.validate_and_add_asset(doc, argument_text, line)
 
+        elif name == "cta-banner":
+            if options is None:
+                self.diagnostics.append(
+                    MissingRequiredOption("cta-banner", "url", line)
```
Comment (rayangler) [inline snooty/parser.py:615]: We may not need this diagnostic / options check. Omitting the url option results in a `DocUtilsParseError` and prints out:

```
ERROR(documents.txt:32ish): "cta-banner" requires the following options: url
```

**Review by rayangler — COMMENTED** [review]
Just one thing from me but looks good!

File: snooty/parser.py (hunk 3 above)
Comment (i80and) [reply snooty/parser.py:615]: Oh, good catch! I forgot that we supported required options via the `required = true` option in rstspec

**Review by i80and — APPROVED** [review]

**Review by rayangler — APPROVED** [review]
LGTM!

**Pull request merged**